cmd/yorkie/user/delete-account.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2024 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package user
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/term"
+
+	"github.com/yorkie-team/yorkie/admin"
+	"github.com/yorkie-team/yorkie/cmd/yorkie/config"
+)
+
+func deleteAccountCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:     "delete-account",
+		Short:   "Delete user account",
+		PreRunE: config.Preload,
+		RunE: func(_ *cobra.Command, args []string) error {
+			fmt.Print("Enter Password: ")
+			bytePassword, err := term.ReadPassword(int(os.Stdin.Fd()))
+			if err != nil {
+				return fmt.Errorf("failed to read password: %w", err)
+			}
+			password = string(bytePassword)
+			fmt.Println()
+
+			fmt.Println(
+				"WARNING: This action is irreversible. Your account and all associated data will be permanently deleted.",
+			)
+
+			fmt.Print("Are you absolutely sure? Type 'DELETE' to confirm: ")
+			var confirmation string
+			if _, err := fmt.Scanln(&confirmation); err != nil {
+				return fmt.Errorf("failed to read confirmation from user: %w", err)
+			}
+
+			if confirmation != "DELETE" {
+				return fmt.Errorf("account deletion aborted")
+			}
+
+			conf, err := config.Load()
+			if err != nil {
+				return err
+			}
+
+			if rpcAddr == "" {
+				rpcAddr = viper.GetString("rpcAddr")
+			}
+
+			if deleteAccountFromServer(conf, rpcAddr, insecure, username, password) == nil {
+				fmt.Println("Your account has been successfully deleted.")
+			}
```

Comment:
I noticed that there's no error handling for failures in the account deletion process. This means that if the deletion fails, the CLI doesn't output any message to the user.
We could add error handling here. This would provide better feedback to users if something goes wrong during the account deletion.